client/src/portal/api/mod.rs

```diff
@@ -342,4 +562,20 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn legacy_decrypt() -> Result<(), Error> {
+        let path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("fixtures")
+            .join("legacy.keyring");
+        let blob = tokio::fs::read(&path).await?;
+        let keyring = LegacyKeyring::try_from(blob.as_slice())?;
+        let password = b"test";
+        let secret: Secret = Vec::from(password).into();
+        let items = keyring.decrypt_items(&secret)?;
+
+        assert_eq!(items.len(), 1);
```

Comment:
Can we have the tests also check that we are reading the item properly?

Reply:
Yes, expanded the test.